File: text_to_csv.cpp
#include <iostream>
#include <fstream>
// #include <string>

using namespace std;

int main (){
    fstream yourfile;
    ofstream ofile;
    string line,amme,a,b,c;
    int i = 0;
    int x,en;
    int currentpercent = 0;
    int whole = 42823960;

    yourfile.open("../TeleDB_light.txt", ios::in | ios::app);  //The path to your file goes here
    ofile.open("../output.csv");
    ofile<<"number_id,tele_id,phone"<<endl;
    while (getline(yourfile,line)){
        amme = line;
        x = amme.find("'id': ");
        if (x != -1){
            en = amme.find(",");
            a = amme.substr(x+6,en-x - 6 );
            amme = amme.substr(en+1);
            x = amme.find("username': ");
            en = amme.find(",");
            b  = amme.substr(x+12,en - x -13);
            x = amme.find("'phone': '");
            en = amme.find("}");
            c = amme.substr(x+10,en - x - 10 - 1);
            ofile<<a<<","<<b<<","<<c<<endl;
            i++;
            if (i > ((currentpercent/100) * whole)){
                cout<<currentpercent<<"% passed..."<<endl;
                currentpercent += 5;
            }
        }
    }
    yourfile.close();
    ofile.close();
}

File: text_to_csv.py
from tqdm import tqdm
import pandas as pd

file = open("../TeleDB_light.txt", "r")
print('opened file')

lines = file.readlines()
print('read lines')
file.close()
out_df = pd.DataFrame(columns =  ['number_id','telegram_id','phone'])
print(lines[0])
print(lines[1])
print(lines[2])
print(lines[3])
print(lines[-4])
print(lines[-3])
print(lines[-2])
print(lines[-1])
print(len(lines))

# for i in tqdm(range(1,len(lines)-2)):
#     x = lines[i].split()
#     num_id = x[1]
#     num_id = num_id[:-1]
#     if (len(x[3]) == 3):
#         tele_id = 'N/A'
#     else:
#         tele_id = x[3]
#         tele_id = tele_id[1:-2]
#     phone_no = x[5]
#     phone_no = phone_no[1:-2]
#     out_df = out_df.append({'number_id':num_id,'telegram_id':tele_id, 'phone' : phone_no},ignore_index=True)

# out_df.to_csv('./teledb.csv')